### DISCUSSION TITLE
[V3] @persist breaks design of the page

### DISCUSSION THREAD

**Discussion by ev3rlost** [livewire/livewire#5936] [1. Help]
Wrapping a page section (`<header>` for example) with `@persist` breaks the design of the page. Is there a workaround I've missed perhaps? Or can you add Tailwind classes on `@persist` itself? I've noticed Livewire adds a div (`<div x-persist="header">`). Or should we use `@persist` only on single elements? Using `x-persist="header"` on the `<header>` element works though.

**Comment by joshhanley** [comment]
@ev3rlost thanks for reporting! Can you share some more details with code examples and screenshots of the issue you're having? Thanks!

**Comment by ev3rlost** [comment]
Okay, I've got a fresh copy of Laravel 10 installed with Livewire V3 (beta-3). 

This is the `app.blade.php` layout file:

```
<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <title>{{ $title ?? '' }}</title>
        @vite('resources/css/app.css')
    </head>
    <body class="bg-gray-100">

        <div class="w-full min-h-screen grid grid-cols-[130px_1fr] grid-rows-[80px_1fr]">

            <header class="col-span-2 bg-gray-800 z-20 flex justify-center items-center text-white">
                Header
            </header>

            <nav class="p-4 bg-white z-10">
                Nav
            </nav>

            <main class="p-4">
                {{ $slot }}
            </main>

        </div>

    </body>
</html>
```

Screenshot of the above code, looks like intended:

![persist_ok](https://github.com/livewire/livewire/assets/20288923/b2ffe999-071a-4993-98ac-44277954854f)

Now if we wrap the `<header>` tag with `@persist`, the layout breaks.

```
@persist('header')
    <header class="col-span-2 bg-gray-800 z-20 flex justify-center items-center text-white">
        Header
    </header>
@endpersist
```

Screenshot of the above code, the layout is broken as Livewire adds an additional div around the header:

![persist_broken](https://github.com/livewire/livewire/assets/20288923/bb42c7a2-cb11-4ef1-a565-0fdde3935929)

The code Livewire renders:

```
<div x-persist="header">
    <header class="col-span-2 bg-gray-800 z-20 flex justify-center items-center text-white">
        Header
    </header>
</div>
```

If we only add `x-persist="header"` on the `<header>` tag itself, it works as intended.

**Comment by joshhanley** [comment]
@ev3rlost yeah that makes sense to me that it breaks the layout as it is injecting a div that has no styles.

I'm not sure what the fix is, whether to allow styles to be passed to `@persist` or whether the documentation should be updated to also support using `x-persist`.

**Reply by ghabriel25** [reply]
@joshhanley Hi, any updates about this? One of use case for this is collapsed menu item (not using alpine collapse). It would be great to persist the styles as well

**Reply by nicolagianelli** [reply]
You can already "handle" that scenario by using x-persist manually on the tag you need it to be:

This:
```
<header x-persist="header" class="col-span-2 bg-gray-800 z-20 flex justify-center items-center text-white">
    Header
</header>
```

Instead of:
```
@persist('header')
    <header class="col-span-2 bg-gray-800 z-20 flex justify-center items-center text-white">
        Header
    </header>
@endpersist
```
